smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -44,6 +44,62 @@ contract MixinTransfer is
   // This is calculated as `keyPrice * transferFeeBasisPoints / BASIS_POINTS_DEN`.
   uint public transferFeeBasisPoints;
 
+  /**
+  * @notice Allows the key owner to share their key (parent key) by
+  * transferring a portion of the remaining time to a new key (child key).
+  * @param _from The owner of the parent key
+  * @param _to The recipient of the shared key
+  * @param _tokenId the key to share
+  * @param _timeShared The amount of time shared
+  */
+  function shareKey(
+    address _from,
+    address _to,
+    uint _tokenId,
+    uint _timeShared
+  ) public
+    onlyIfAlive
+    hasValidKey(_from)
+    onlyKeyOwnerOrApproved(_tokenId)
+  {
+    require(_to != address(0), 'INVALID_ADDRESS');
+
+    Key storage fromKey = keyByOwner[_from];
+    Key storage toKey = keyByOwner[_to];
+    uint IdTo = toKey.tokenId;
+    uint time;
+    // get the remaining time for the origin key
+    uint timeRemaining = fromKey.expirationTimestamp - block.timestamp;
+    // get the transfer fee
+    uint fee = getTransferFee(_from, _timeShared);
+
+    // ensure that we don't try to share too much
+    if(_timeShared.add(fee) < timeRemaining) {
+      time = _timeShared;
+      uint timePlusFee = _timeShared.add(fee);
+      // deduct time from parent key, including transfer fee
+      _timeMachine(_tokenId, timePlusFee, false);
+    } else {
+      fee = getTransferFee(_from, timeRemaining);
+      time = timeRemaining.sub(fee);
+      fromKey.expirationTimestamp = block.timestamp; // Effectively expiring the key
+      emit ExpireKey(_tokenId);
+    }
+
+    if (toKey.tokenId == 0) {
+      _assignNewTokenId(toKey);
+      _recordOwner(_to, IdTo);
+    }
+    // add time to new key
+    _timeMachine(IdTo, time, true);
+    // trigger event
+    emit Transfer(
+      address(0), // This is a creation or time-sharing
```

Comment:
Hmm is from address(0) best here?  maybe we should use the from account?

Reply:
Yeah, could do. I guess it depends on the recipient. If they don't have a key, a new key is being created for them. If they already have a key, they just get a top-up. 
I really don't like either option...but equally fine with using either.

Reply:
I think doing it once or twice makes sense.  If it's a new key, emit from 0 as the spec requires that on creation.  Then always emit from -> to as well at the end.

Reply:
That works for me.